pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -131,13 +130,19 @@ public void process()
         }
       }
 
-      // Create/modify/remove star-trees if required.
-      processStarTrees(indexDir);
-
-      // Perform post-cleanup operations on the index handlers. This should be called after processing the startrees
+      // Perform post-cleanup operations on the index handlers.
       for (IndexHandler handler : indexHandlers) {
         handler.postUpdateIndicesCleanup(segmentWriter);
       }
+      segmentWriter.save();
+    }
+
+    // Startree creation will load the segment again, so we need to close and re-open the segment writer to make sure
+    // that the other required indices (e.g. forward index) are up-to-date.
+    try (SegmentDirectory.Writer segmentWriter = _segmentDirectory.createWriter()) {
+      // Create/modify/remove star-trees if required.
+      processStarTrees(indexDir);
+      _segmentDirectory.reloadMetadata();
 
       // Add min/max value to column metadata according to the prune mode.
       // For star-tree index, because it can only increase the range, so min/max value can still be used in pruner.
```

Comment:
Is there a necessity to process startree index before setting min/max?

Reply:
Good question. I was simply keeping the same order. I will try out moving the min/max update to happen along with the index generation. I think that startree index should be independent of column based indices and not impact to min/max metadata. I will follow up on this.

Reply:
changed the order